components/core/src/clp/DictionaryConcepts.hpp

```diff
@@ -0,0 +1,262 @@
+     */
+    {
+        entry.get_placeholder_info(placeholder_ix, placeholder_ref)
+    } -> std::same_as<size_t>;
+
+    /**
+     * @return The dictionary Id for this logtype.
+     */
+    {
+        entry.get_id()
+    } -> std::same_as<logtype_dictionary_id_t>;
+};
+
+/**
+ * Requirement for the variable dictionary entry interface.
+ * @tparam VariableDictionaryEntryType The type of the variable dictionary entry.
+ */
+template <typename VariableDictionaryEntryType>
+concept VariableDictionaryEntryReq = requires(VariableDictionaryEntryType entry) {
+    /**
+     * @return The dictionary Id for this variable.
+     */
+    {
+        entry.get_id()
+    } -> std::same_as<variable_dictionary_id_t>;
+};
+
+/**
+ * Requirement for the logtype dictionary reader interface.
+ * @tparam LogTypeDictionaryReaderType The type of the logtype dictionary reader.
+ * @tparam LogTypeDictionaryEntryType The type of the entries in the logtype dictionary reader.
+ */
+template <
+        typename LogTypeDictionaryReaderType,
+        typename LogTypeDictionaryEntryType = typename LogTypeDictionaryReaderType::entry_t>
+concept LogTypeDictionaryReaderReq = requires(
+        LogTypeDictionaryReaderType reader,
+        std::string_view logtype,
+        bool ignore_case,
+        std::unordered_set<LogTypeDictionaryEntryType const*>& entries
+) {
+    /**
+     * Gets entries matching a given logtype.
+     * @param logtype
+     * @param ignore_case Whether the lookup should be case insensitive.
+     * @return A vector of entries matching the given logtype.
+     */
+    {
+        reader.get_entry_matching_value(logtype, ignore_case)
+    } -> std::same_as<std::vector<LogTypeDictionaryEntryType const*>>;
+
+    /**
+     * Gets entries matching a wildcard string.
+     * @param logtype A wildcard search string.
+     * @param ignore_case Whether the search should be case insensitive.
+     * @param entries A hash set in which to store the found entries.
+     */
+    {
+        reader.get_entries_matching_wildcard_string(logtype, ignore_case, entries)
+    } -> std::same_as<void>;
+
+    requires std::
+            same_as<typename LogTypeDictionaryReaderType::dictionary_id_t, logtype_dictionary_id_t>;
+
+    requires std::
+            same_as<typename LogTypeDictionaryReaderType::entry_t, LogTypeDictionaryEntryType>;
+};
+
+/**
+ * Requirement for the variable dictionary writer interface.
+ * @tparam VariableDictionaryWriterType The type of the variable dictionary writer.
+ */
+template <typename VariableDictionaryWriterType>
+concept VariableDictionaryWriterReq = requires(
+        VariableDictionaryWriterType writer,
+        std::string_view value,
+        variable_dictionary_id_t& id_ref
+) {
+    /**
+     * Adds the given variable to the dictionary if it doesn't exist.
+     * @param value
+     * @param id_ref The Id of the variable matching the given entry.
+     * @return Whether this call resulted in inserting a new entry or not.
```

Comment:
```suggestion
     * @param id_ref Returns the entry ID of the given variable.
     * @return Whether this call resulted in inserting a new entry.
```
We don't need to add "or not" in a "whether" return statement.